icu4c/source/i18n/number_modifiers.cpp

```diff
@@ -110,7 +145,7 @@ void ConstantAffixModifier::getParameters(Parameters& output) const {
     UPRV_UNREACHABLE_EXIT;
 }
 
-bool ConstantAffixModifier::semanticallyEquivalent(const Modifier& other) const {
+bool ConstantAffixModifier::strictEquals(const Modifier& other) const {
```

Comment:
Just out of curiosity, why the name change?  And is there a "lenient" or "approximate" equals function?

(**UPDATED TO ADD:**  Oh, I see.  You still have a `semanticallyEquivalent()` method as well.  You commented at the declaration site about how they're different, but you might also want to say something there about who needs one and who needs the other and why.)

While I'm at it, what are your thoughts on using `operator==()` for this?  Is it already being used to do something else, or does it have other problems that make it impossible or inconvenient?

Reply:
I err away from `operator==` because that doesn't work in Java and I think it's one of the things that causes bugs like this in the first place, since it's not obvious during a port whether `==` invokes a special impl.

Reply:
Yeah, I thought your reasoning would be something like that.  That's fine with me; I have no objection to what you're doing here.